src/modules/network/network.py

```diff
@@ -13,20 +13,16 @@ class Network:
     def __init__(self):
         self.client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         self.client.settimeout(5.0)  # may raise socket.timeout exception
-        self.server = "10.0.0.137"  # WARNING: hardcode & need to be consistenet with server.py
-        self.port = 5555
-        self.server_addr = (self.server, self.port)
-        self.connect()
 
-    def connect(self):
-        try:
-            self.client.connect(self.server_addr)
-            if s.decode_connect_response(self.client.recv(2048)):
-                logger.info("Connection established.")
-            else:
-                logger.error("Connection failed to establish.")
-        except:
-            pass
+    def connect(self, ip: str):
+        logger.info("Connecting to %s", ip)
+        host, port = ip.split(":")
+        self.client.connect((host, int(port)))
+
+        if s.decode_connect_response(self.client.recv(2048)):
+            logger.info("Connection established.")
+        else:
```

Comment:
just a note here - i'll change it later so that there won't be the else condition since serializer will throw exception